### DISCUSSION TITLE
Support "TestFlight Internal Only" Build

### DISCUSSION THREAD

**Discussion by YuantongL** [fastlane/fastlane#21543] [Ideas]
This year new to the AppStoreConnect, there's the "TestFlight Internal Only" build, it will be great to have that option mapped to fastlane `upload_to_testflight`.

Context: [What’s new in App Store Connect](https://developer.apple.com/videos/play/wwdc2023/10117/) at 4:37

**Comment by YuantongL** [comment]
https://developer.apple.com/documentation/appstoreconnectapi/buildaudiencetype

**Comment by Greensource** [comment]
@YuantongL I'm also interesting by these feature but I guess we need to propose a solution to make this subject go forward.
I will try to understand this part of the source code

**Comment by YuantongL** [comment]
I have a draft [PR](https://github.com/fastlane/fastlane/pull/29859) to add this option. @Greensource
